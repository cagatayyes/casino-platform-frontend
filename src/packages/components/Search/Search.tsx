import React, { useRef, useState } from 'react';
import classNames from 'classnames';
import { JSX } from 'react';
import { Button } from 'casino/components/Button/Button';

export interface ISearchProps extends React.ComponentProps<'input'> {
	placeholder?: string
	className?: string
	inputClassName?: string
	clearClassName?: string
	iconClassName?: string
	cancelClassName?: string
	value?: string
	cancelText?: string
	searchIconColor?: string
	clearIconColor?: string
	onChange?(e: React.ChangeEvent<HTMLInputElement> | string): void
	onCancel?(e: React.MouseEvent<HTMLButtonElement>): void
	onClear?(e?: React.MouseEvent<HTMLButtonElement>): void
}

const Search = ({
	placeholder,
	value,
	cancelText,
	className,
	iconClassName,
	inputClassName,
	cancelClassName,
	clearClassName,
	searchIconColor,
	clearIconColor,
	onChange,
	onCancel,
	onClear,
	...props
}: ISearchProps) : JSX.Element => {
	const inputRef:React.RefObject<HTMLInputElement> = useRef<HTMLInputElement>(null);
	const [inputValue, setInputValue] = useState<string>('');
	const [isFocus, setIsFocus] = useState<boolean>(false);

	const inputFocus = (event: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>) : void => {
		if ((event.target as HTMLDivElement)?.tagName !== 'BUTTON') {
			inputRef.current?.focus();
			setIsFocus(true);
		}
	};

	const clearValue = () : void => {
		onChange?.call(this, '');
		inputRef.current?.focus();
		onClear?.call(this);
		setInputValue('');
	};

	const handleChange = (e : React.ChangeEvent<HTMLInputElement>) : void => {
		onChange?.call(this, e.target.value);
		setInputValue(e.target.value);
	};

	return (
		<div
			className={classNames(
				`inline-flex relative flex-row p-[10px] items-center ${isFocus ? 'border-[#FFB800] bg-[#FFFFFF33] text-[#ffffff]': ''}`,
				className,
			)}
			onClick={inputFocus}
			onKeyDown={inputFocus}
		>
			<div
				className={classNames(
					'text-[#008751] mr-[5px] flex-grow-0 flex items-center',
					iconClassName,
				)}
			>
				<span
					className="w-[16px] h-[16px] material-symbols-outlined mt-[3px]"
					style={{ color: searchIconColor, fontSize:'16px' }}>
					search
				</span>
			</div>
			<div className="relative flex flex-1 items-center">
				<input
					ref={inputRef}
					value={inputValue}
					type="text"
					placeholder={placeholder}
					className={classNames(
						`bg-[transparent] outline-none w-full h-full text-[11px] pr-[24px] text-[11px] ${isFocus ? 'text-[#ffffff] ' : 'text-[#6C7086]'}`,
						inputClassName,
					)}
					onChange={(e) : void => {
						handleChange(e);
						setIsFocus(true);
					}}
					onBlur={() : void => {
						setIsFocus(false);
					}}
					{...props}
				/>
				{inputValue && <Button
					className={classNames(
						'absolute right-0 top-0 text-[#008751] transition-all flex items-center h-full',
						clearClassName,
						{
							'opacity-100': !!value,
						},
					)}
					onClick={clearValue}
					noStyle
					label={<span
						className="material-symbols-outlined w-[16px] h-[16px] mt-[3px] ml-[5px]"
						style={{ color: clearIconColor, fontSize: '12px' }}
					>
						close
					</span>}
				/>}
			</div>
			{cancelText && <div className="flex-grow-0 pl-[25px] pr-[6px] items-center flex">
				<Button
					name="cancel"
					className={classNames(
						'text-[#008751] text-[11px] font-medium font-[500]',
						cancelClassName,
					)}
					onClick={onCancel}
					noStyle
					label={<span
						className='material-symbols-outlined'
						style={{ fontSize: '16px' }}
					>close</span>}
				/>
			</div>
			}
		</div>
	);
};

export default Search;
