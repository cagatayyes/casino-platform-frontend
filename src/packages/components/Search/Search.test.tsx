import React from 'react';
import Search from './Search';
import { render } from '@testing-library/react';


describe('Search', () => {
	it('should display Search component', () => {
		const setInputValue = jest.fn();

		const searchText = 'Search';
		React.useState = jest.fn().mockReturnValue([searchText, setInputValue]);

		const { container } = render(<Search
			placeholder='Search Here'
			className='flex w-full'
			inputClassName='bg-[linear-gradient(0deg, #E9F0FD, #E9F0FD)]'
			searchIconColor='#000'
			clearIconColor='#000'
			onChange={jest.fn()}
		/>);

		expect(container).toMatchSnapshot();

	});
});
