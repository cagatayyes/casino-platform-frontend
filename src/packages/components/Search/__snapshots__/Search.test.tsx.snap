// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should display Search component 1`] = `
<div>
  <div
    class="inline-flex relative flex-row p-[10px] items-center  flex w-full"
  >
    <div
      class="text-[#008751] mr-[5px] flex-grow-0 flex items-center"
    >
      <span
        class="w-[16px] h-[16px] material-symbols-outlined mt-[3px]"
        style="color: rgb(0, 0, 0); font-size: 16px;"
      >
        search
      </span>
    </div>
    <div
      class="relative flex flex-1 items-center"
    >
      <input
        class="bg-[transparent] outline-none w-full h-full text-[11px] pr-[24px] text-[11px] text-[#6C7086] bg-[linear-gradient(0deg, #E9F0FD, #E9F0FD)]"
        placeholder="Search Here"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
